src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataResultSetLatestIT.java

```diff
@@ -32,4 +35,27 @@ public void testGetObjectNotSupported() throws SQLException {
       }
     }
   }
+
+  /** Added in > 3.17.1 */
+  @Test
+  public void testObjectColumn() throws SQLException {
+    try (Connection connection = getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "CREATE OR REPLACE TABLE TABLEWITHOBJECTCOLUMN ("
+              + "    col3 OBJECT("
+              + "      str VARCHAR,"
+              + "      num NUMBER(38,0)"
+              + "      )"
+              + "   )");
+      DatabaseMetaData metaData = connection.getMetaData();
+      try (ResultSet resultSet =
+          metaData.getColumns(
+              connection.getCatalog(), connection.getSchema(), "TABLEWITHOBJECTCOLUMN", null)) {
+        assertTrue(resultSet.next());
+        assertEquals("OBJECT", resultSet.getObject(6)); // column type
```

Comment:
can we get a column by name to avoid a comment?

Reply:
Done.